cmd/cli/plugin/cluster/upgrade.go

```diff
@@ -323,13 +341,62 @@
 func getClusterTKRNameFromClusterLabels(clusterLabels map[string]string) (string, error) {
 	tkrLabelExists := false
 	tkrName := ""
-	// TODO: Once the TKGs and TKGm label for TKR are same, update this
-	tkrName, tkrLabelExists = clusterLabels["run.tanzu.vmware.com/tkr"] // TKGs
+	tkrName, tkrLabelExists = clusterLabels[runv1.LabelTKR]
 	if !tkrLabelExists {
-		tkrName, tkrLabelExists = clusterLabels["tanzuKubernetesRelease"] // TKGm
+		tkrName, tkrLabelExists = clusterLabels[LegacyClusterTKRLabel] // legacy TKGm
 	}
 	if !tkrLabelExists {
 		return "", errors.New("failed to get cluster TKr name from cluster labels")
 	}
 	return tkrName, nil
 }
+
+func getValidTKRVersionFromClusterForUpgrade(cluster *capiv1.Cluster, tkrName string) (string, error) {
+	tkrVersion := strings.ReplaceAll(tkrName, "---", "+")
+	// If the existing TKR on the cluster and the one requested are same
+	// continue the upgrade using the given TKR
+	if clusterTKR, err := getClusterTKRNameFromClusterLabels(cluster.Labels); err == nil && clusterTKR == tkrName {
```

Comment:
This is too complicated. 

Looks like it could be replaced with
```suggestion
	if cluster.Labels[runv1.LabelTKR] == tkrName {
```

If so, then seems we don't need to check the new label in `getClusterTKRNameFromClusterLabels()`

Reply:
Not sure I understand your comment. We need to check for both legacy cluster(label is `tanzuKubernetesRelease`) and updated label on classy clusters(`runv1.LabelTKKR` -- run.tanzu.vmware.com/tkr).

Reply:
Nevermind, we are going to call this with legacy-style clusters, too.